config/routes.rb

```diff
@@ -3,4 +3,16 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  # Homepage route
+  get 'users', to: 'home#index', as: :users
+
+  # User profile route
+  get 'users/:id', to: 'users#show', as: :user
+
+  # User posts route
+  get 'users/:user_id/posts', to: 'posts#index', as: :user_posts
+
```

Comment:
Kindly implement a corresponding route for the user & post indexes previously mentioned.